### PR TITLE
fix: create rules if score is external

### PR DESCRIPTION
**Pull request by tikhanovichA**
Disable external property depends on what outcome variable was used 

Depends on: https://github.com/oat-sa/tao-item-runner-qti-fe/pull/414

https://github.com/user-attachments/assets/d8834f49-f4e6-4580-b571-f53c5a5d548e

### CODE REVIEW

File: views/js/qtiCreator/plugins/panel/outcomeEditor.js [hunk 1]
```diff
@@ -96,10 +96,14 @@ define([
     function renderListing(item, $outcomeEditorPanel) {
         const readOnlyRpVariables = getRpUsedVariables(item);
         const scoreMaxScoreVisible = features.isVisible('taoQtiItem/creator/interaction/response/outcomeDeclarations/scoreMaxScore');
+        let scoreExternalScored = _.get(_.find(item.outcomes, function (outcome) {
```
Comment (viktar-dzmitryieu-tao) [inline views/js/qtiCreator/plugins/panel/outcomeEditor.js:99]: can't find where we reassign it, maybe `const`?

**Review by viktar-dzmitryieu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [X] New code is not subject to concurrency issues (if applicable)
- [X] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful
- [X] Pull request's target is not `master`

File: views/js/qtiCreator/plugins/panel/outcomeEditor.js [hunk 2]
```diff
@@ -108,11 +112,31 @@ define([
...
+                        otherOutcome.attributes.externalScored !== externalScoredOptions.none;
+                });
+
+                if (hasOtherExternalScored) {
+                    externalScored.none.selected = true;
+                    externalScored.human.selected = false;
+                    externalScored.externalMachine.selected = false;
+                    externalScoredDisabled = 1;
+                }
+            }
```
Comment (Karol-Stelmaczonek) [inline views/js/qtiCreator/plugins/panel/outcomeEditor.js:134]: Check my proposal.
```suggestion
function setExternalScoredToNone() {
    externalScored.none.selected = true;
    externalScored.human.selected = false;
    externalScored.externalMachine.selected = false;
    externalScoredDisabled = 1;
}

function hasExternalScoredOutcome(outcomes) {
    return outcomes.some(outcome => 
        outcome.attributes &&
        outcome.attributes.identifier !== 'SCORE' &&
        outcome.attributes.externalScored &&
        outcome.attributes.externalScored !== externalScoredOptions.none
    );
}

function shouldSetExternalScoredToNone() {
    if (id !== 'SCORE') {
        return scoreExternalScored && scoreExternalScored !== externalScoredOptions.none;
    }
    return hasExternalScoredOutcome(item.outcomes);
}

if (shouldSetExternalScoredToNone()) {
    setExternalScoredToNone();
}
```

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by KirylHatalski — APPROVED** [review]

**Pull request merged**